lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
@bajtos As alternate approach, I tried this in model builder

  // Add the id property
  if (idInjection) {
    // Set up the id property
    // updateOnly property is added to indicate that this property will appear in the model for update related operations
    // but and not for create operation.
    if (ModelClass.definition.settings.forceId !== false) {
      ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
    } else {
      ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
    }
  }
}
Unfortunately, this doesn't seem to work either if user explicitly sets forceId = false in model file, for e.g  book.json file.  ModelClass.definition.settings.forceId logic seems to look at the base model's (Model) setting rather than the book model's setting.